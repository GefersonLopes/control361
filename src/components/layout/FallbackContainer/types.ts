export interface FallbackContainerProps {
  className?: string;
  children: React.ReactNode;
}
