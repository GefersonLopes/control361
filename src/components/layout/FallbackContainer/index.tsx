import clsx from "clsx";
import React from "react";

import type { FallbackContainerProps } from "./types";

const FallbackContainer: React.FC<FallbackContainerProps> = ({
  className,
  children,
}) => (
  <div
    className={clsx(
      "h-[518px] overflow-hidden rounded-2xl border border-slate-700/60 flex items-center justify-center",
      className,
    )}
  >
    {children}
  </div>
);

export default FallbackContainer;
