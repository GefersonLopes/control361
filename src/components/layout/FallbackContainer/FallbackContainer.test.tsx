import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import FallbackContainer from ".";

describe("FallbackContainer component", () => {
  it("renders a div with default classes and children", () => {
    render(
      <FallbackContainer>
        <span>Content</span>
      </FallbackContainer>,
    );
    const container = screen.getByText("Content").parentElement;
    expect(container).toBeInTheDocument();
    expect(container?.tagName).toBe("DIV");
    [
      "h-[518px]",
      "lg:h-[70vh]",
      "overflow-hidden",
      "rounded-2xl",
      "border",
      "border-slate-700/60",
      "flex",
      "items-center",
      "justify-center",
    ].forEach((cls) => {
      expect(container).toHaveClass(cls);
    });
  });

  it("merges a custom className prop", () => {
    render(
      <FallbackContainer className="extra-class another-class">
        <p>Test</p>
      </FallbackContainer>,
    );
    const container = screen.getByText("Test").parentElement;
    expect(container).toHaveClass("extra-class");
    expect(container).toHaveClass("another-class");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <FallbackContainer className="snap-class">
        <div>Snap</div>
      </FallbackContainer>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
