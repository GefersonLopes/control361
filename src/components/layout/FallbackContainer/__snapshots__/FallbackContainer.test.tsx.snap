// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FallbackContainer component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-[518px] lg:h-[70vh] overflow-hidden rounded-2xl border border-slate-700/60 flex items-center justify-center snap-class"
  >
    <div>
      Snap
    </div>
  </div>
</DocumentFragment>
`;
